swiftide-core/src/indexing_traits.rs

```diff
@@ -547,14 +588,79 @@ pub trait SimplePrompt: Debug + Send + Sync + DynClone {
 
 dyn_clone::clone_trait_object!(SimplePrompt);
 
+#[derive(Debug, Clone)]
+pub struct ReliableLanguageModel<P: Clone> {
+    inner: P,
+    config: BackoffConfiguration,
+}
+
+impl<P: Clone> ReliableLanguageModel<P> {
+    pub fn new(client: P, config: BackoffConfiguration) -> Self {
+        Self {
+            inner: client,
+            config,
+        }
+    }
+
+    fn strategy(&self) -> backoff::ExponentialBackoff {
+        backoff::ExponentialBackoffBuilder::default()
+            .with_initial_interval(Duration::from_secs(self.config.initial_interval_sec))
+            .with_multiplier(self.config.multiplier)
+            .with_max_elapsed_time(Some(Duration::from_secs(self.config.max_elapsed_time_sec)))
+            .with_randomization_factor(self.config.randomization_factor)
+            .build()
+    }
+}
+
+#[async_trait]
+impl<P: SimplePrompt + Clone> SimplePrompt for ReliableLanguageModel<P> {
```

Comment:
I'm a bit stuck on the name here. I think if it wraps the client with backoff, we should call it that. `Reliable` feels a bit like jibe / wishful thinking.

Reply:
You're right, I also confused reliability with resilience, would `ResilientLanguageModel` be better? Or `LanguageModelWithBackoff`?

Reply:
Yeah I think the latter works. Can you think of. a better place to put it than in traits?